scrunch.rb

```diff
@@ -1,17 +1,16 @@
+# coding: utf-8
 # -*-Ruby-*-
 
-def get_metadata(input_file)
-  unprocessed = `AtomicParsley \"#{input_file}\" -t`
-  array_lines = unprocessed.split(/\n/)
-  array_complicated = []
-  array_lines.each { |i|
-    array_complicated << i.split(": ")
+def get_metadata(file)
+  atoms = `AtomicParsley \"#{file}\" -t`
+  atoms.each_line.reduce({}) { |hash, line|
```

Comment:
The `#each_line` method gives us an iterator, which is a construct for enumerating. This is easier on memory than using `#lines`, since we don't actually need an array of lines in memory at any point.
`reduce` is like `map`, but it passes along the result of the previous block. The empty hash argument is used as the initial value.